Source/WTF/wtf/NativePromise.h

```diff
@@ -942,8 +963,8 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
     }
 
 public:
-    template<typename SettleFunction>
-    auto whenSettled(RefCountedSerialFunctionDispatcher& targetQueue, SettleFunction&& settleFunction, const Logger::LogSiteIdentifier& callSite = DEFAULT_LOGSITEIDENTIFIER)
+    template<typename Dispatcher, typename SettleFunction>
```

Comment:
, I don't see why you need this when ScriptExecutionContextNativePromiseDispatcher is a RefCountedSerialFunctionDispatcher